src/script/stack.rs

```diff
@@ -250,24 +250,24 @@ impl ExecutionStack {
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, ScriptError> {
-        let mut items = Vec::new();
+        let mut stack = ExecutionStack { items: Vec::new() };
         let mut byte_str = bytes;
         while !byte_str.is_empty() {
             match StackItem::read_next(byte_str) {
                 Some((item, b)) => {
-                    items.push(item);
+                    stack.push(item)?;
                     byte_str = b;
                 },
                 None => return Err(ScriptError::InvalidInput),
             }
         }
-        Ok(ExecutionStack { items })
+        Ok(stack)
     }
 
     /// Pushes the item onto the top of the stack. This function will only error if the new stack size exceeds the
     /// maximum allowed stack size, given by [MAX_STACK_SIZE]
     pub fn push(&mut self, item: StackItem) -> Result<(), ScriptError> {
-        if self.size() >= MAX_STACK_SIZE {
+        if self.size() > MAX_STACK_SIZE {
```

Comment:
```suggestion
        if self.size() >= MAX_STACK_SIZE {
```
because we're about to increase the size by one, which would exceed MAX SIZE if it were equal.

Reply:
This one was actually more about the docs/error message (I decreased MAX_STACK_SIZE by one)
In the comments it says if it _exceeds_ [MAX_STACK_SIZE] which was 256 - meaning that it can be 256, but in reality 255 is the maximum it can contain.

Reply:
The original looks consistent to me. 

"This function will only error if the **new stack size** exceeds the maximum allowed stack size, given by [MAX_STACK_SIZE]"

MAX_STACK_SIZE should be whatever. And that's the maximum stack size, not MAX_STACK_SIZE+1. I figured you changed MAX_STACK_SIZE for some other reason.

Reply:
"This function will only error if the new stack size exceeds the maximum allowed stack size of 256 (previous value)". - to exceed 256, the new stack size must be 257 or greater. Was a pedantic change so no problems with changing it back :) will do so

Reply:
Ah I'm finally with you, the MAX_STACK_SIZE should be 255 with the `>=` - damn off-by-one errors!